lib/galaxy/util/script.py

```diff
@@ -26,30 +26,43 @@ def main(argv=None):
     if argv is None:
         argv = sys.argv[1:]
     args = _arg_parser().parse_args(argv)
-    kwargs = _app_properties(args)
+    kwargs = app_properties_from_args(args)
     action = args.action
     action_func = ACTIONS[action]
     action_func(args, kwargs)
 
 
-def _app_properties(args):
-    # FIXME: you can use galaxy.util.path.extensions for this
-    config_file = args.config_file or find_config_file(args.app)
+def app_properties_from_args(args, legacy_config_override=None):
+    config_file = config_file_from_args(args, legacy_config_override=legacy_config_override)
     app_properties = load_app_properties(config_file=config_file, config_section=args.config_section)
     return app_properties
 
 
+def config_file_from_args(args, legacy_config_override=None):
+    # FIXME: you can use galaxy.util.path.extensions for this
+    config_file = legacy_config_override or args.config_file or find_config_file(getattr(args, "app", "galaxy"))
+    return config_file
+
+
+def populate_config_args(parser):
+    # config and config-file respected because we have used different arguments at different
+    # time for scripts.
+    parser.add_argument("-c", "--config-file", "--config",
+                        default=os.environ.get('GALAXY_CONFIG_FILE', None),
+                        help="Galaxy config file (defaults to config/galaxy.ini or config/galaxy.yml)")
+    parser.add_argument("--config-section",
```

Comment:
I've made the changes to the methods you requested but I've left this argument in. I think on balance @natefoo and I think it is a good idea and you do not. Are you -1 with it in place and going to force a change or can you live with it here?

I've showed a lot of restraint in not adding ``--config_directory`` and ``--galaxy_root`` as arguments if that helps any 😃 - my impulse is to make this even more adaptable.

Reply:
I know that as developers we feel the urge of exposing as many options as possible to the users, but I think that before doing it there should be a use case. Almost no script needed this option till now, so probably there is no use case?

Reply:
Some of these scripts didn't even allow configuration of the config file, most of them wouldn't respect configuration via environment variables - they would only work under a very narrow range of deployment scenarios. A best practice for deployment (ideal for both package management and containerized deployments) would have strong, clear separation between app directories and configuration directories with different owners and permissions - so one wouldn't place configuration files into config/ directly. That deployment wouldn't work with some of these scripts - it doesn't mean there aren't use cases for doing that - it is the best practice. I'm not arguing that there is a stunning use case here for --config-section just that it is a fallacy that these scripts didn't do something in the past therefore we shouldn't do it or there isn't a use case.

Since galaxy supports the use case of serving multiple instances of galaxy from the same configuration file and some of the scripts feature explicit support for this (galaxy-main, manage_tool_dependencies), all the scripts should. Anything else is a bug since some scripts support a configuration and others don't IMO. How about this - since you aren't advocating gutting the infrastructure that allows this just to reduce the cognitive load on users - what if we set the help to ``argparse.SUPPRESS``. People who want the advanced options have it available to them but we don't populate every help message with it.

Reply:
That seems a good compromise, didn't know about `help=argparse.SUPPRESS` ! And sorry for being a pain!